src/modules/external_services/external_ip.py

```diff
@@ -0,0 +1,45 @@
+from .exceptions import ExternalServiceFailure, ExternalProviderFailure
+
+import requests
+from collections import namedtuple
+
+def generic_text_http_provider(url):
+    try:
+        return requests.get(url).text
+    except requests.ConnectionError as e:
+        raise ExternalProviderFailure(e)
+
+ExtIPProvider = namedtuple('ExternalIPProvider', [
+    'handler',
+    'priority' # it means: secure, no logs, etc.
+])
+
+CanHazIP = ExtIPProvider(handler=generic_text_http_provider('https://canhazip.com'), priority=0)
+Ipify = ExtIPProvider(handler=generic_text_http_provider('https://api.ipify.org'), priority=1)
+
+DEFAULT_PROVIDERS = [CanHazIP, Ipify]
+
+class ExternalIPService:
```

Comment:
I think this class can be a general `BaseExternalService` with failover in it so it can be reused

Reply:
Do you mean like, I should go ahead and generalize it for any type of results?

Reply:
I felt like I could do it, but I was wondering if it was a really good idea, but if you want I can do the modifications and make it a `external_base_service.py` and derive `external_ip.py` from it.

Reply:
It feels right to me, I think it will save some code duplication in future.
Let's go with a `base_external_service.py` with `BaseExternalService` class.

Reply:
Alright, I'll perform those last changes tomorrow then and fix conflicts